jenkins-integration/jenkins-jobs/common/scripts/jenkins/pipeline.groovy

```diff
@@ -51,6 +61,36 @@ def get_server_era(pe_version) {
     }
 }
 
+def get_oss_server_era(oss_version) {
+    // TODO: eventually we will probably want to do something more sophisticated
+    //  here; currently only support 'latest'/'master'/'stable' OSS puppetserver,
+    //  and 'latest' agent
+    if (["latest", "master", "stable"].contains(oss_version)) {
+        return [type: "oss",
+                service_name: "puppetserver",
+                version: oss_version,
+                agent_version: "latest",
```

Comment:
Might want to do `agent_version` as a job-specific param at the `Jenkinsfile` level?  Defaulting to `latest` if not specified seems fine but I could imagine us wanting to easily different versions of agent and server together.  Seems good for now, though.